### ISSUE TITLE
Onedir mode : Pyinstaller embeds system libraries

### ISSUE THREAD

**Issue by ptitoliv** [pyinstaller/pyinstaller#2728]
New to pyinstaller world, I try to make some executables from my python code using PyInstaller in one-dir mode. The binary works but there is a thing I don't understand : When generating the EXE, Pyinstaller copied lots of system libraries into the dist directory where the EXE has been generated.

I browsed a little bit the code and it seems that when a BINARY type is detected, a copy is done. Is it the normal behaviour ? Shouldn't the BINARY type elements which seem to be system dynamic librarires excluded of the copy operation when doing the COLLECT ?

I'm asking the question because, in the pyinstaller doc, it is written 

> PyInstaller does not include libraries that should exist in any installation of this OS. For example in Linux, it does not bundle any file from /lib or /usr/lib, assuming these will be found in every system.

I must precise that in my case, libs are located in /lib64 directory but for me it should be the same behaviour than for /lib or /usr/lib.

**Comment by htgoebel** [comment]
Looks like the documentation needs to be more precise. PyInstaller copies dynlibs which are to be expected to *not* exist on the target system. This includes thinks like Python modules implemented as dynlib, openssl, QT, gtk, etc.

Does this answer your question?

**Comment by ptitoliv** [comment]
Hello,

Thanks for your quick answer. Indeed, this is not I understood from reading the doc. It is definitely clearer explained like that.

However, I would be very interested if on the onedir mode (and maybe also on the onefile mode), if we could use the system libraries instead of embedding the one from the build system. Of course, it shouldn't be the default behavior but a specific enable with a command line option. 

But maybe it is not the correct issue to discuss about it since it's now related about the documentation. Would you be interested to know my point of view about this potential feature on a dedicated issue ?

**Comment by htgoebel** [comment]
> Would you be interested to know my point of view about this potential feature on a dedicated issue ?

Yes, pleas, If you want to discuss this, please open a new ticket.
